File: typescript/src/address/format/formatAddress.ts
import { RestApiFormatAddress } from "../../server/address/format/restApiFormatAddress";

export type FormatAddress = {
    addressLine1: string;
    addressLine2: string;
    addressLine3: string;
    locality: string;
    region: string;
    postalCode: string;
    country: string;
}

export function restApiResponseToFormatAddress(response: RestApiFormatAddress): FormatAddress {
    return {
        addressLine1: response.address_line_1??"",
        addressLine2: response.address_line_2??"",
        addressLine3: response.address_line_3??"",
        locality: response.locality??"",
        region: response.region??"",
        postalCode: response.postal_code??"",
        country: response.country??""
    }
}

File: csharp/DVSClient/DVSClient/address/format/FormatAddress.cs
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format
{
    public class FormatAddress
    {
        public string AddressLine1 { get; }
        public string AddressLine2 { get; }
        public string AddressLine3 { get; }
        public string Locality { get; }
        public string Region { get; }
        public string PostalCode { get; }
        public string Country { get; }

        public FormatAddress(RestApiFormatAddress apiAddress)
        {
            AddressLine1 = apiAddress.AddressLine1 ?? string.Empty;
            AddressLine2 = apiAddress.AddressLine2 ?? string.Empty;
            AddressLine3 = apiAddress.AddressLine3 ?? string.Empty;
            Locality = apiAddress.Locality ?? string.Empty;
            Region = apiAddress.Region ?? string.Empty;
            PostalCode = apiAddress.PostalCode ?? string.Empty;
            Country = apiAddress.Country ?? string.Empty;
        }
    }
}